pkg/diagnostics/http_tracing.go

```diff
@@ -58,6 +58,12 @@ func HTTPTraceMiddleware(next fasthttp.RequestHandler, appID string, spec config
 			}
 		}
 
+		// Check if response has traceparent header and add if absent
+		if !checkResponseHeader(&ctx.Response, traceparentHeader) {
```

Comment:
as discussed, check for tracestateheader also, there are similar methods in utils...

Reply:
Shouldn't we discard tracestate if traceparent is absent as per w3C trace-context spec [here](https://www.w3.org/TR/trace-context/#no-traceparent-received)?